pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/XPathPanelController.java

```diff
@@ -112,9 +118,66 @@ public void initialize(URL location, ResourceBundle resources) {
                            // Reevaluate XPath anytime the expression or the XPath version changes
                            .or(xpathVersionProperty().changes())
                            .subscribe(tick -> parent.refreshXPathResults());
+
+        xpathExpressionArea.plainTextChanges()
+                           .filter(t -> (t.getInserted().matches("[a-zA-Z]")) || t.getInserted().matches("/"))
+                           .subscribe(t -> {
+                               try {
+                                   if (t.getInserted().equals("/")) {
+                                       autoComplete(xpathExpressionArea.getText().substring(xpathExpressionArea.getText().indexOf("/")));
+                                   } else if (!t.getInserted().equals("/")) {
+                                       autoComplete(xpathExpressionArea.getText().substring(xpathExpressionArea.getText().lastIndexOf("/")));
+                                   }
+                               } catch (IOException e) {
+                                   e.printStackTrace();
+                               } catch (ClassNotFoundException e) {
+                                   e.printStackTrace();
+                               }
+                           });
+    }
+
+    private void autoComplete(String s) throws IOException, ClassNotFoundException {
+
+        autoCompletePopup = new ContextMenu();
+        List<MenuItem> resultToDisplay = new ArrayList<>();
+
+        String language = parent.getLanguageVersion().getName().replaceAll("[0-9]", "").replaceAll("//s", "").toLowerCase().trim();
+        XPathSuggestions xPathSuggestions = new XPathSuggestions("net.sourceforge.pmd.lang." + language + ".ast");
```

Comment:
give that `XPathSuggestions` knows about this package (it filters out on `XPathSuggestions .evaluateXpathSuggestions`, I'd consider just having the constructor receive the language name, and have the rest of the logic moved there. This way, there is a single class that needs to care about what packages are expected to contain AST nodes for a given language.